### PR TITLE
Split upgrade/install helm docs

### PR DESCRIPTION
**Pull request by jacob-delgado**
Helm installation documentation as well as upgrades (w/o canaries) are all done in a single document. Split it out similarly to the other documentation used for istioctl

This will also allow upgrades to be done as part of automated tests.

[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by jacob-delgado** [comment]
cc @nrjpoddar The plan is to have this be similar to istioctl. I can also test this as its a different page under istio.io testing

File: content/en/boilerplates/helm-backup.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+---
+Before upgrading Istio in your cluster, we recommend creating a backup of your
+custom configurations, and restoring it from backup if necessary:
+
+{{< text bash >}}
+$ kubectl get istio-io --all-namespaces -oyaml > "$HOME"/istio_resource_backup.yaml
```
[the remark this answers is not in the document]
Comment (jacob-delgado) [reply content/en/boilerplates/helm-backup.md:7]: This is in the original doc. https://istio.io/latest/docs/setup/install/helm/#upgrading-using-helm

It's from CustomResourceDefinition per https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#categories and defined in istio in https://github.com/istio/istio/blob/42d25d8fe9ecd8fe50fd9261047f28c0544622a4/manifests/charts/base/crds/crd-all.gen.yaml#L17

This is the first time I've seen that in the docs. Pretty useful command.

**Comment by jacob-delgado** [comment]
> Not related to this but can we make [deploy-preview-9098--preliminary-istio.netlify.app/latest/docs/setup/install/helm/#verifying-the-installation](https://deploy-preview-9098--preliminary-istio.netlify.app/latest/docs/setup/install/helm/#verifying-the-installation) step be `istioctl verify-install`? When I did Helm work `istioctl` command was not working with Helm based installs.

It still doesn't work with Helm based installs

```shell
$ ./bin/istioctl verify-install
1 Istio control planes detected, checking --revision "default" only
1 Istio injectors detected
Error: Istio present but verify-install needs an IstioOperator or manifest for comparison. Supply flag --filename <yaml>
```

File: content/en/docs/setup/upgrade/helm/index.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+{{< boilerplate helm-hub-tag >}}
+
+{{< boilerplate helm-prereqs >}}
+
+## Upgrade steps
+
+Change directory to the root of the release package and then
+follow the instructions below.
+
+{{< boilerplate helm-jwt-warning >}}
```
[the remark this answers is not in the document]
Comment (jacob-delgado) [reply content/en/docs/setup/upgrade/helm/index.md:27]: Where do you think we should move it? If they doing an upgrade they will have already done a helm based install. I'm fine with removing it, but it seems like there are some that would forget about this. We don't mention to the user that they should use an overrides file. Maybe we should in the install doc and then also reference that here?

File: content/en/boilerplates/helm-backup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrjpoddar) [reply content/en/boilerplates/helm-backup.md:7]: Thanks! I didn't know this existed.

File: content/en/docs/setup/upgrade/helm/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrjpoddar) [reply content/en/docs/setup/upgrade/helm/index.md:27]: Sounds good.

File: content/en/docs/setup/upgrade/helm/index.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+### Create a backup
+
+{{< boilerplate helm-backup >}}
+
+### Migrating from non-Helm installations
+
+{{< boilerplate helm-migration-nonhelm >}}
+
+### Canary upgrade (recommended)
```
[the remark this answers is not in the document]
Comment (nrjpoddar) [reply content/en/docs/setup/upgrade/helm/index.md:37]: Let's keep it canary for now..

**Review by nrjpoddar — APPROVED** [review]
Approving with minor outstanding suggestion which is not a blocker for this PR.

File: content/en/docs/setup/upgrade/helm/index.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+description: Upgrade and configure Istio for in-depth evaluation.
+weight: 27
+keywords: [kubernetes,helm]
+owner: istio/wg-environments-maintainers
+icon: helm
+test: no
+---
+
+Follow this guide to upgrade and configure an Istio mesh using
+[Helm](https://helm.sh/docs/) for in-depth evaluation.  This guy assumes you have already performed an
```
[the remark this answers is not in the document]
Comment (ericvn) [reply content/en/docs/setup/upgrade/helm/index.md:13]: You missed this change.

File: content/en/docs/setup/upgrade/helm/index.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+Change directory to the root of the release package and then
+follow the instructions below.
+
+{{< boilerplate helm-jwt-warning >}}
+
+### Create a backup
+
+{{< boilerplate helm-backup >}}
+
+### Migrating from non-Helm installations
```
Comment (ericvn) [inline content/en/docs/setup/upgrade/helm/index.md:33]: Would you migrate from a non-helm installation during upgrade? Also the template say to `re-install Istio using Helm as described above` which isn't true anymore (it's on another page).

File: content/en/docs/setup/upgrade/helm/index.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+    {{< text bash >}}
+    $ helm upgrade istio-ingress manifests/charts/gateways/istio-ingress \
+        -n istio-system
+    $ helm upgrade istio-egress manifests/charts/gateways/istio-egress \
+        -n istio-system
+    {{< /text >}}
+
+## Uninstall
+
```
Comment (ericvn) [inline content/en/docs/setup/upgrade/helm/index.md:116]: Tried looking at the other upgrade pages and not sure I see uninstall there (it is for the canary control plane). Should we just point to the uninstall on the install page and not duplicate?

File: content/en/docs/setup/upgrade/helm/index.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+You can uninstall Istio and its components by uninstalling the charts
+installed above.
+
+1. List all the Istio charts installed in `istio-system` namespace:
+
+    {{< text bash >}}
+    $ helm ls -n istio-system
```
Comment (ericvn) [inline content/en/docs/setup/upgrade/helm/index.md:123]: I do note we don't show a list here. We do on the install page. Not sure if we should/shouldn't. If we just point to the install page, this comment doesn't matter.

File: content/en/docs/setup/upgrade/helm/index.md (hunk 4 above)
Comment (ericvn) [inline content/en/docs/setup/upgrade/helm/index.md:13]: The change from guy to guide was missed.

Comment (jacob-delgado) [reply content/en/docs/setup/upgrade/helm/index.md:13]: Fixed

File: content/en/docs/setup/upgrade/helm/index.md (hunk 5 above)
Comment (jacob-delgado) [reply content/en/docs/setup/upgrade/helm/index.md:33]: Removed this section from the upgrade template. I agree with you.

File: content/en/docs/setup/upgrade/helm/index.md (hunk 6 above)
Comment (jacob-delgado) [reply content/en/docs/setup/upgrade/helm/index.md:116]: Yeah. Pushed a link to it.

**Review by ericvn — APPROVED** [review]

**Pull request merged**